packages/core/lib/command-utils.js

```diff
@@ -201,6 +203,37 @@ const runCommand = async function (command, options) {
   return await command.run(options);
 };
 
+const processHelpInput = inputStrings => {
+  //When we think user is trying to access help incorrectly, provide them with some helpful information.
+  if (inputStrings.length > 2 && ["help", "--help"].includes(inputStrings[1])) {
+    const help = require("./commands/help").meta;
+    console.log(
+      "Error: please use the following syntax for accessing help information\n",
+      "\n               ",
+      help.help.usage
+    );
+    process.exit();
+  }
+
+  // when `truffle --help <cmd>` is used, convert inputStrings to run as `truffle help <cmd>`
+  if (inputStrings.length > 1 && inputStrings[0] === "--help") {
```

Comment:
maybe it would also be better to test for `inputStrings.length === 2` since we already know `inputStrings.length` is not greater than 2 from line 208

Reply:
hmm. Not sure if I'm following. 
Line 208 checks for when the `inputStrings.length` is greater than 2, and `help` is exactly at `inputStrings[1]` which is not a syntax order truffle is looking to support. 

And line 219 checks for examples like `truffle --help migrate` and ` truffle --help db serve` where the `inputString.length` is bigger than 1 and possibly bigger than 2 with the use of subCommand.